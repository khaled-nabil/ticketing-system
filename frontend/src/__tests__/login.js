import React from 'react';
import {mount, shallow} from 'enzyme';
import Login from '../components/login';

const setup = propOverrides => {
    const props = Object.assign({
        fetchToken: jest.fn(({email, password})=> true)
    }, propOverrides);

    const wrapper = mount(<Login {...props} />);

    return {
        props,
        wrapper,
        form: wrapper.find("form")
    }
};

describe('Login Render', () => {
    it('Login renders correctly', () => {
        const component = shallow(<Login />);
        expect(component).toMatchSnapshot();
    });
    it('form submits to fetch token',()=>{
        const component = setup();
        component.form.simulate('submit');
        expect(component.props.fetchToken).toHaveBeenCalledTimes(1);
        expect(component.props.fetchToken).toHaveBeenCalledWith( {"email": "test@test.com", "password": "123"});

    })
});
