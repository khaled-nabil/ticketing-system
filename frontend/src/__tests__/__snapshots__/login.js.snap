// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Render Login renders correctly 1`] = `ShallowWrapper {}`;
